src/Chapter3.hs

```diff
@@ -971,6 +1064,38 @@ implement the following functions:
 🕯 HINT: to implement this task, derive some standard typeclasses
 -}
 
+data WeekDay = Sunday
+             | Monday
+             | Tuesday
+             | Wednesday
+             | Thursday
+             | Friday
+             | Saturday
+             deriving (Show, Eq)
+
+isWeekend :: WeekDay -> Bool
+isWeekend Saturday = True
+isWeekend Sunday = True
+isWeekend _ = False
+
+nextDay :: WeekDay -> WeekDay
+nextDay Sunday = Monday
+nextDay Monday = Tuesday
+nextDay Tuesday = Wednesday
+nextDay Wednesday = Thursday
+nextDay Thursday = Friday
+nextDay Friday = Saturday
+nextDay Saturday = Sunday
+
+daysToParty :: WeekDay -> Int
+daysToParty Sunday = 5
+daysToParty Monday = 4
+daysToParty Tuesday = 3
+daysToParty Wednesday = 2
+daysToParty Thursday = 1
+daysToParty Friday = 0
+daysToParty Saturday = 6
```

Comment:
These functions don't use standard functions of standard typeclasses. I would recommend looking into `Bounded` and `Enum` typeclasses, derive them and try to see what of their methods you can use :slightly_smiling_face: